music21/midi/translate.py

```diff
@@ -515,6 +517,15 @@ def midiEventsToChord(eventList, ticksPerQuarter=None, inputM21=None):
     pitches = []
     volumes = []
 
+    def handleEvent(event, pitches, volumes, thisChord):
+        p = pitch.Pitch()
+        p.midi = event.pitch
+        v = volume.Volume(velocity=event.velocity)
+        v.velocityIsRelative = False  # velocity is absolute coming from MIDI
+        pitches.append(p)
+        volumes.append(v)
+        thisChord.editorial['channels'].append(event.channel)
```

Comment:
since channels is MIDI specific and not respected for output, can we make it `importedFromMidiChannels`?  or something like that?  

Are there cases where notes/chords in a single part will be imported from multiple channels?  Should this be put on the Part object instead?

Reply:
I assumed in the reporter's use case--with the Jammy guitar with different fingerings on each note represented by different channels--there would be a stream of notes in a single track all with different channels, and this is what was being preserved. I'll ask!

Reply:
If this is too much overhead then maybe we don't want this in core, and then @nightscape can just fork and merge this branch.

Reply:
Pitches are quite lightweight -- they use slots, etc. to keep them really memory efficient.  The editorial was something I debated putting on but decided to do so because it only adds a single byte per pitch if it's None, but I think it will double the size of pitches if it's declared on every object, and then people parsing very large MIDI files will notice.

What really needs to happen is a complete refactor of the MIDI translate into two files toMidi and fromMidi with __init__.py going to objects.py and an object-oriented translator that can have keyword settings on what types of quantization, whether to add channel information, etc. but that's a MUCH larger project.